frontend/src/chat/Chat.tsx

```diff
@@ -0,0 +1,154 @@
+import React, { useState, useEffect, useRef } from "react";
+import Card from "react-bootstrap/Card";
+import { Message, MessageData } from "./Message";
+import { api } from "../services/apiService";
+import socketIOClient from "socket.io-client";
+
+const chat_server_url = "http://localhost:3001";
+
+function createSocket() {
+    return api.jwt
+        .getToken()
+        .then((token: string) => {
+            return io.connect(chat_server_url, {
+                forceNew: true,
+                query: "token=" + token
+            });
+        })
+        .catch(() => {
```

Comment:
I don't think the `catch` is necessary  if you do nothing

Reply:
Strange, it seems like our versions are different, any way this should be okay now